spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -202,6 +203,20 @@ public void register(@PathVariable("type") ApplicationType type, @PathVariable("
 			@PathVariable("version") String version,
 			@RequestParam("uri") String uri, @RequestParam(name = "metadata-uri", required = false) String metadataUri,
 			@RequestParam(value = "force", defaultValue = "false") boolean force) {
+
+		// compare existing default app and deny registration if base path of uri's
+		// don't match as registering app with different version can only differ by version
+		AppRegistration defaultApp = appRegistryService.getDefaultApp(name, type);
```

Comment:
I think we should move this logic out of the controller and into a `validate` method on `appRegistryService` ?